lib/checkleakautovar.cpp

```diff
@@ -678,14 +743,60 @@ void CheckLeakAutoVar::functionCall(const Token *tok, VarInfo *varInfo, const Va
 
             // Is variable allocated?
             if (!isnull && (!af || af->arg == argNr))
-                changeAllocStatus(varInfo, allocation, tok, arg);
-        } else if (Token::Match(arg, "%name% (")) {
-            const Library::AllocFunc* allocFunc = _settings->library.dealloc(arg);
-            VarInfo::AllocInfo alloc(allocFunc ? allocFunc->groupId : 0, VarInfo::DEALLOC);
-            if (alloc.type == 0)
-                alloc.status = VarInfo::NOALLOC;
-            functionCall(arg, varInfo, alloc, allocFunc);
+                changeAllocStatus(varInfo, allocation, tokName, arg);
+        }
+        // Check smart pointer
+        else if (Token::Match(arg, "auto_ptr|unique_ptr|shared_ptr < %type%")) {
+
```

Comment:
I usually don't use 2 empty blank lines in function bodies. I don't think it is _wrong_ but it might be unconventional.